3MI0800270/2023-02-28/zad2.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+using namespace std;
+
+struct Tuple {
+    int first;
+    int second;
+
+    void input() {
```

Comment:
The name of this method is rather confusing - something like `read` will make a lot more sense :)) In addition to that - you've fixed your `read` to only using the [std::cin](https://www.geeksforgeeks.org/cin-in-c/). This is in fact an example of poor separation of concerns. In the given case you are concerned on how to read the `first` & `second` but you're also concerned on the source from which you're reading - `cin`. If you're interested - you can check [this page](https://en.wikipedia.org/wiki/Dependency_injection) and learn more about the topic :))